components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -265,12 +266,20 @@ export class ParsedCommand {
                     // Positional parameter.
                     this.position++;
                     // eslint-disable-next-line no-loop-func
-                    const field = fields.find((f: AppField) => f.position === this.position);
+                    let field = fields.find((f: AppField) => f.position === this.position);
                     if (!field) {
-                        return this.asError(this.intl.formatMessage({
-                            id: 'apps.error.parser.no_argument_pos_x',
-                            defaultMessage: 'Unable to identify argument.',
-                        }));
+                        field = fields.find((f) => f.position === -1 && f.type === AppFieldTypes.TEXT);
+                        if (!field || this.values[field.name]) {
```

Comment:
The rest field is a normal field with name. So the `--text` field in this case can be set as:
`/command --text "hello world"`
or
`/command hello world`

The point of this check here is to avoid this:
`/command --text "hello world" hello world`

Reply:
I am a little confused, if `--text`'s type is set to "rest-of-the-line", then `/command --text "hello world" hello world` should have `text` set to `"\"hello world\"" hello world"`, am I confused?

Reply:
`/command --text "hello world" hello world` will return a parsing error, since the flag --text has the value `hello world` before reaching the rest field.
`/command "hello world" hello world` will be parsed ok, and assing the flag --text the value `"hello world" hello world"`.

A more clarifying example of why I want the first one to fail is the following:
`/command --text "hello world" --number 1 hello world`